src/utils/tweetUpdates.ts

```diff
@@ -0,0 +1,89 @@
+import { wikiActivities } from '../services/types/activityResult'
+import { TwitterApi } from 'twitter-api-v2'
+
+export const shortenAddress = (address: string) => {
+  const match = address.match(
+    /^(0x[a-zA-Z0-9]{4})[a-zA-Z0-9]+([a-zA-Z0-9]{4})$/,
+  )
+  if (!match) return address
+  return `${match[1]}…${match[2]}`
+}
+
+export default class WikiUpdatesTweeter {
+  private client: TwitterApi
+
+  constructor() {
+    this.client = new TwitterApi({
+      appKey: process.env.TWITTER_API_KEY as string,
+      appSecret: process.env.TWITTER_API_SECRET as string,
+      accessToken: process.env.TWITTER_ACCESS_TOKEN as string,
+      accessSecret: process.env.TWITTER_ACCESS_SECRET as string,
+    })
+  }
+
+  async tweetWikiActivity(activity: wikiActivities, url: string) {
+    // TWEET ELEMENTS
+    const wikiTitle = activity.content[0].title
+    let editorName = activity.user.profile?.username || activity.user.id
+    if (editorName.startsWith('0x')) {
+      editorName = shortenAddress(editorName)
+    }
+    const wikiRelatedTwitterAccount = `(@${
+      activity.content[0].metadata.find(m => m.id === 'twitter_profile')?.value
+    }) `
+      .replace('https://twitter.com/', '')
+      .replace('(@)', '')
+    const editorTwitterAccount = `@${
+      activity.user.profile?.links?.find(l => l.twitter)?.twitter
+    }`.replace('https://twitter.com/', '')
+    const wikiURL = `${url.replace('/wiki', '/revision')}${activity.id}`
+    const hashTags = [
+      'Wiki',
+      ...activity.content[0].categories.map(c => c.title),
+      ...activity.content[0].tags.map(t => t.id),
+    ].map(tag =>
+      `#${tag}`
+        .split(' ')
+        .map(w => w[0].toUpperCase() + w.substring(1))
+        .join(''),
+    )
+
+    // BUILDING TWEET TEXT WITH FORMATTING
```

Comment:
comments in a 80 lines function its a smell that you could split it into several smaller private functions to give readability

Reply:
Ohh thanks, i will split them to private methods. this should make it easy to read